src/system-plugins/input-controller/public/js/input-controller.js

```diff
@@ -16,6 +16,23 @@
 
   var recordPluginNeeded = true;
 
+  //Mapping of axis to inversions actions
```

Comment:
Instead of a hardcoded map, seems like you should be able to update the registration method so that it wraps the function that is called in an inversionHandler function:

```javascript
function wrapInInversionCheck(fn){
  return function(value){
    if (axisForThiscontrolIsInverted()){
      value=value * -1;
    }
    fn(value);
  }
}

register(controls,fn){
fn = wrapInInversionCheck(fn);
...
}
```